CMakeLists.txt

```diff
@@ -242,17 +242,21 @@ endif()
 
 include(TryAppendCFlags)
 if(MSVC)
-  # Keep the following commands ordered lexicographically.
+  # For both cl and clang-cl compilers.
   try_append_c_flags(/W3) # Production quality warning level.
+  # Eliminate deprecation warnings for the older, less secure functions.
+  add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+else()
+  try_append_c_flags(-Wall) # GCC >= 2.95 and probably many other compilers.
+endif()  
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+  # Keep the following commands ordered lexicographically.
```

Comment:
I'd prefer to set `/W3` here and `-Wall` with the other `try_append_c_flags` for consistency, as I don't see why it needs to be special-cased.

Reply:
According to the [docs](https://clang.llvm.org/docs/UsersManual.html#id11), the `/Wall` option in clang-cl, which is equivalent to `-Wall`, enables Clang's core `-Weverything` option.

We use `/W3`, which enables Clang's core `-Wall` option.

Reply:
Wow. Confirmed: https://godbolt.org/z/WEr3qGeox

That's... hugely unexpected. It really doesn't give me much faith in the proper forwarding of other options :(